crates/core/c8y_smartrest/src/event.rs

```diff
@@ -0,0 +1,109 @@
+                    || current_timestamp.format(&Rfc3339),
+                    |timestamp| timestamp.format(&Rfc3339)
+                )?
+            );
+
+            Ok(smartrest_message)
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use assert_matches::assert_matches;
+    use csv::Error;
+    use serde::Deserialize;
+    use test_case::test_case;
+    use thin_edge_json::event::ThinEdgeEventData;
+    use time::macros::datetime;
+
+    #[test_case(
+        ThinEdgeEvent {
+            name: "click_event".into(),
+            data: Some(ThinEdgeEventData {
+                message: Some("Someone clicked".into()),
+                time: Some(datetime!(2021-04-23 19:00:00 +05:00)),
+            }),
+        },
+        "400,click_event,\"Someone clicked\",2021-04-23T19:00:00+05:00"
+        ;"event translation"
+    )]
+    #[test_case(
+        ThinEdgeEvent {
+            name: "click_event".into(),
+            data: Some(ThinEdgeEventData {
+                message: None,
+                time: Some(datetime!(2021-04-23 19:00:00 +05:00)),
+            }),
+        },
+        "400,click_event,\"\",2021-04-23T19:00:00+05:00"
+        ;"event translation without message"
+    )]
+    #[test_case(
+        ThinEdgeEvent {
+            name: "click_event".into(),
+            data: Some(ThinEdgeEventData {
+                message: Some("Someone, clicked, it".into()),
+                time: Some(datetime!(2021-04-23 19:00:00 +05:00)),
+            }),
+        },
+        "400,click_event,\"Someone, clicked, it\",2021-04-23T19:00:00+05:00"
+        ;"event translation with commas in message"
+    )]
+    fn check_event_translation(event: ThinEdgeEvent, expected_smartrest_msg: &str) {
+        let result = serialize_event(event);
+
+        assert_eq!(result.unwrap(), expected_smartrest_msg);
+    }
+
+    #[derive(Debug, Deserialize)]
+    struct SmartRestEvent {
+        pub code: i32,
+        pub name: String,
+        pub message: Option<String>,
+        pub time: Option<String>,
+    }
+
+    #[test]
+    fn event_translation_empty_json_payload_generates_timestamp() {
+        let event = ThinEdgeEvent {
+            name: "empty_event".into(),
+            data: Some(ThinEdgeEventData {
+                message: None,
+                time: None,
+            }),
+        };
+
+        let smartrest_message = serialize_event(event).unwrap();
+        let mut reader = csv::ReaderBuilder::new()
+            .has_headers(false)
+            .from_reader(smartrest_message.as_bytes());
+        let mut iter = reader.deserialize();
+        let result = iter.next();
+
+        assert!(result.is_some());
+        let smartrest_event: SmartRestEvent = result.unwrap().unwrap();
+        assert_eq!(smartrest_event.code, 400);
+        assert_eq!(smartrest_event.name, "empty_event".to_string());
+        assert_eq!(smartrest_event.message, None);
+        assert_matches!(smartrest_event.time, Some(_))
+    }
```

Comment:
Would this test benefit from using return error and `?` instead of `unwrap`?
```suggestion
    use anyhow::Result;

    fn event_translation_empty_json_payload_generates_timestamp() -> Result<()>{
        let event = ThinEdgeEvent {
            name: "empty_event".into(),
            data: Some(ThinEdgeEventData {
                message: None,
                time: None,
            }),
        };

        let smartrest_message = serialize_event(event).unwrap();
        let mut reader = csv::ReaderBuilder::new()
            .has_headers(false)
            .from_reader(smartrest_message.as_bytes());
        let mut iter = reader.deserialize();
        let result = iter.next();

        assert!(result.is_some());
        let smartrest_event: SmartRestEvent = result??;
        assert_eq!(smartrest_event.code, 400);
        assert_eq!(smartrest_event.name, "empty_event".to_string());
        assert_eq!(smartrest_event.message, None);
        assert_matches!(smartrest_event.time, Some(_))
    }
```